From d481d0c69bba7d690813ea7a72e2d29dc7c9bb69 Mon Sep 17 00:00:00 2001
From: Joseph Hwang <jzh@fb.com>
Date: Wed, 22 Jun 2016 08:55:41 -0700
Subject: [PATCH] HBASE-16080 Fix flakey test
 TestTableBasedReplicationSourceManager.cleanupFailoverQueues(). Instead of
 running the primary test in a separate thread and hoping it finishes in time,
 just run the test in the primary thread.

---
 .../hbase/replication/regionserver/TestReplicationSourceManager.java   | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/hbase-server/src/test/java/org/apache/hadoop/hbase/replication/regionserver/TestReplicationSourceManager.java b/hbase-server/src/test/java/org/apache/hadoop/hbase/replication/regionserver/TestReplicationSourceManager.java
index 4b278bb..f60982e 100644
--- a/hbase-server/src/test/java/org/apache/hadoop/hbase/replication/regionserver/TestReplicationSourceManager.java
+++ b/hbase-server/src/test/java/org/apache/hadoop/hbase/replication/regionserver/TestReplicationSourceManager.java
@@ -345,8 +345,7 @@ public abstract class TestReplicationSourceManager {
     NodeFailoverWorker w1 =
         manager.new NodeFailoverWorker(server.getServerName().getServerName(), rq1, rp1, new UUID(
             new Long(1), new Long(2)));
-    w1.start();
-    w1.join(10000);
+    w1.run();
     assertEquals(1, manager.getWalsByIdRecoveredQueues().size());
     String id = "1-" + server.getServerName().getServerName();
     assertEquals(files, manager.getWalsByIdRecoveredQueues().get(id).get(group));
-- 
2.8.0-rc2

